proxy/Main.cc

```diff
@@ -476,6 +476,18 @@ check_config_directories(void)
   }
 }
 
+namespace {
+  // Values with names that have this prefix are exempted from being considered "unrecognized".
+  const char PLUGIN_CONFIG_RECORD_PREFIX[] = "proxy.config.plugin.";
```

Comment:
This is the root of the problem in that if arbitrary config names can be used by plugins, you can't do validity checks. The consensus at the bug scrub was it was reasonable to restrict plugin configuration values to that subtree.

I could move the check after plugin initialization so that if a plugin does call {{TSMgmtStringCreate}} it doesn't generate a warning.